pkg/ipam/node.go

```diff
@@ -209,8 +209,16 @@ func (n *Node) getMinAllocate() int {
 }
 
 // getMaxAllocate returns the maximum-allocation setting of an AWS node
-func (n *Node) getMaxAllocate() int {
-	return n.resource.Spec.IPAM.MaxAllocate
+func (n *Node) getMaxAllocate(scopedLog *logrus.Entry) int {
+	instanceMax := n.ops.GetMaximumAllocatableIPv4(scopedLog)
+	if n.resource.Spec.IPAM.MaxAllocate > 0 {
+		if n.resource.Spec.IPAM.MaxAllocate > instanceMax {
+			scopedLog.Warningf("max-allocate (%d) is higher than the instance type limits (%d)", n.resource.Spec.IPAM.MaxAllocate, instanceMax)
```

Comment:
Do we want the user to do something about this? What could the user do? Should we add "Consider configuring --option-foo ...." to the end of this log message?

Reply:
I reckon that to receive this message the user will already have had to set the flag/option 🤔
Perhaps we could add some more precise potential consequences though such as `this could lead to ip allocation overflows`?

Reply:
If I follow right, the problem here is there are instance limits and there is the user-specified max, and the user has wrongly configured the max too high so we hit the instance limit first. Is that right? In this case the recommended action for the user would be to lower the max option.

My overall point is that printing a log that says "warning: here is a fact" with no solution is noisy and requires extrapolation from the user to figure out what they can do about the problem.

Reply:
Looks like this message could also use the `WithFields()` function.

Reply:
I have decorated all functions with fields 👍 As for the logging itself, up to you. I completely get your point and am happy to remove it if you guys feel this is too noisy or irrelevant 👌